### DISCUSSION TITLE
Mac - cursor-style doesn't take effact

### DISCUSSION THREAD

**Discussion by sunshine-syz** [ghostty-org/ghostty#3004] [Q&A] [answered]
I set the `cursor-style = block` but it doesn't change the cursor style. Unless I run `zsh` to start a session again? iTerm2 doesn't have this issue.
I wonder does anyone know this issue or is this a bug?

**Comment by vinoff** [comment]
Same issue here.

**Comment by brendanbeck62** [comment]
Same issue here on macos. I am using a default config. When I launch ghostty, my cursor is `bar` style even though the default config is `cursor=block`. If I launch tmux or run `zsh` again as OP said, then it uses the correct `block` style.

**Reply by brendanbeck62** [reply]
@sunshine-syz is there a reason you closed this without an explaination? Seems like a bug to me, is it a duplicate?

**Comment by sunshine-syz** [comment, marked as answer]
I solved the issue by setting this
`shell-integration-features = no-cursor`

**Reply by brendanbeck62** [reply]
Thanks, fixed it for me!
Note, this called out in the docs and I missed it:

https://ghostty.org/docs/config/reference#cursor-style
> Note that shell integration will automatically set the cursor to a bar at a prompt, regardless of this configuration. You can disable that behavior by specifying shell-integration-features = no-cursor or disabling shell integration entirely.
